src/plugins/gs-plugin-external-appstream.c

```diff
@@ -94,24 +105,102 @@ create_tmp_file (GsPlugin *plugin,
 	return FALSE;
 }
 
+static char *
+get_modification_date (const char *file_path)
```

Comment:
Agreed; this code is going to be run in a thread and I don't think setlocale() is a good idea at all.